ui/src/shared/components/TimeSeries.tsx

```diff
@@ -314,6 +316,17 @@ class TimeSeries extends Component<Props & WithRouterProps, State> {
   }
 
   private shouldReload(prevProps: Props) {
+    if (this.props.hasUpdatedTimeRangeInVEO) {
+      return false
+    }
+
+    if (
+      prevProps.hasUpdatedTimeRangeInVEO &&
+      !this.props.hasUpdatedTimeRangeInVEO
```

Comment:
Do you think our query caching will actually do this for us? without having to indicate whether user has updated time range in VEO? Let's think it through

Reply:
SO this condition is used specifically to trigger an update after a timeRange selection has been made, AND after the VEO has been closed. Without this, toggling the timeRange (even with the initial cache) will cause us to requery the dashboard, so this is to make sure that that doesn't happen until the configuration process has been completed

Reply:
but we'd prevent the re querying of the dashboard because we can tell from the router that the VEO is displayed. Then when we close the VEO, we can reissue queries, and if time range has not changed the results will be returned from the cache. if the time range has changed we won't find a match in the cache, and will issue a query.

Reply:
I think the difference is that we still want to reissue the queries if the timerange selected is the same timerange selected. So if a user has `-5m` selected, they scroll through the dropdown, select other options, then settle back on `-5m`, the query technically will not have changed, but the event should be registered and should update the other cells

Reply:
you're right.

Reply:
I think we might want to update all cells whenever any change has been applied to a cell. or else the time frames will be out of sync.